### PR TITLE
IAM 330

### PR DESCRIPTION
**Pull request by shipperizer**
IAM-330: introduce otel for tracing
IAM-326: separate logic from http handler for extra package

- feat: IAM-330 - introduce otel tracing
- feat: wire up tracer inside status endpoint as a dummy example
- ci: update ossf action
- feat: IAM-330 - use otelhttp middleware
- feat: IAM-326 - separate logic and http handling in extra package
- feat: wire up service in extra package

TODO:
- [x] rebase on top of #45 once merged
- [x] rebase on top of #63  once merged

### CODE REVIEW

**Review by nsklikas — APPROVED** [review]

File: internal/tracing/tracer.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+	if cfg.JaegerEndpoint != "" {
+		exporter, err = jaeger.New(
+			jaeger.WithCollectorEndpoint(
+				jaeger.WithEndpoint(cfg.JaegerEndpoint),
+			),
+		)
+	} else {
+		exporter, err = stdouttrace.New(
+			stdouttrace.WithPrettyPrint(),
+		)
```
Comment (nsklikas) [inline internal/tracing/tracer.go:98]: Shouldn't this be configurable as well? (someone may not want tracing enabled at all)

Comment (nsklikas) [inline internal/tracing/tracer.go:98]: Shouldn't this be configurable as well? (someone may not want tracing enabled at all)

**Review by nsklikas — COMMENTED** [review]
Other than a little comment it looks good

File: internal/tracing/tracer.go (hunk 1 above)
Comment (shipperizer) [reply internal/tracing/tracer.go:98]: added ticket to sort this out https://warthogs.atlassian.net/browse/IAM-353

**Review by nsklikas — APPROVED** [review]

**Pull request merged**